### PR TITLE
Globally add VKB_DEBUG for debug builds

### PR DESCRIPTION
**Pull request by asuessenbach**
## Description

With #404, an inconsistency with the definition of VKB_DEBUG was introduced, that led to a crash on closing a sample and corruptions on rotating the scene.
Globally defining VKB_DEBUG for debug builds fixes this.

Builds and runs on Win10, using Nvidia hardware

## General Checklist:

Please ensure the following points are checked:

- [x] My code follows the [coding style](https://github.com/KhronosGroup/Vulkan-Samples/tree/master/CONTRIBUTING.md#Code-Style)
- [x] I have reviewed file [licenses](https://github.com/KhronosGroup/Vulkan-Samples/tree/master/CONTRIBUTING.md#Copyright-Notice-and-License-Template)
- [x] I have updated the header Copyright to reflect the current year (CI build will fail if Copyright is out of date)
- [x] I have commented any added functions (in line with Doxygen)
- [x] I have commented any code that could be hard to understand
- [x] My changes do not add any new compiler warnings
- [x] My changes do not add any new validation layer errors or warnings
- [x] I have used existing framework/helper functions where possible
- [ ] My changes build on Windows, Linux, macOS and Android. Otherwise I have [documented any exceptions](https://github.com/KhronosGroup/Vulkan-Samples/tree/master/CONTRIBUTING.md#General-Requirements)
- [x] My changes do not add any regressions
- [x] I have tested every sample to ensure everything runs correctly
- [x] This PR describes the scope and expected impact of the changes I am making

## Sample Checklist

If your PR contains a new or modified sample, these further checks must be carried out *in addition* to the General Checklist:
- [ ] I have tested the sample on at least one compliant Vulkan implementation
- [ ] If the sample is vendor-specific, I have [tagged it appropriately](https://github.com/KhronosGroup/Vulkan-Samples/tree/master/CONTRIBUTING.md#General-Requirements)
- [ ] I have stated on what implementation the sample has been tested so that others can test on different implementations and platforms
- [ ] Any dependent assets have been merged and published in downstream modules
- [ ] For new samples, I have added a paragraph with a summary to the appropriate chapter in the [samples readme](./../samples/README.md)

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -41,6 +41,9 @@ endif()
 
 cmake_minimum_required(VERSION 3.16)
 
+# globally add VKB_DEBUG for the debug build
+add_compile_definitions($<$<CONFIG:DEBUG>:VKB_DEBUG>)
```
Comment (tomadamatkinson) [inline CMakeLists.txt:45]: Global compile definitions are generally a bad practice. But this seems to be the simplest solution.

Did you try adding the compile definition in the `add_project` function?

Comment (asuessenbach) [reply CMakeLists.txt:45]: I mostly agree, regarding global definitions. But this error occurred due to having VKB_DEBUG defined only locally, and not for every project. Therefore I think, having it globally is the right way here.

I did not try to add it to the `add_project` function. And even if I had, at least the framework project is not instantiated via that function.

Comment (tomadamatkinson) [reply CMakeLists.txt:45]: Cool, I am happy if others are. As discussed on slack, this works as the name of the define is so unique

**Review by gpx1000 — APPROVED** [review]

**Review by SaschaWillems — APPROVED** [review]

**Review by gary-sweet — APPROVED** [review]

**Comment by marty-johnson59** [comment]
Merging w/ 3 approvals

**Pull request merged**